test/contconic.jl

```diff
@@ -503,5 +503,105 @@ function contconictest(solver::MOI.AbstractSolver, ε=Base.rtoldefault(Float64))
+            #
+            #      (x1,x2,x3) in C^3_q
+            #      X in C_sdp
+
+            m = MOI.SolverInstance(solver)
+
+            X = MOI.addvariables!(m, 6)
+            @test MOI.getattribute(m, MOI.NumberOfVariables()) == 6
+            x = MOI.addvariables!(m, 3)
+            @test MOI.getattribute(m, MOI.NumberOfVariables()) == 9
+
+            cX = MOI.addconstraint!(m, MOI.VectorVariablewiseFunction(x), MOI.PositiveSemidefiniteConeTriangle(3))
+            cx = MOI.addconstraint!(m, MOI.VectorVariablewiseFunction(X), MOI.SecondOrderCone(3))
+
+            c1 = MOI.addconstraint!(m, MOI.ScalarAffineFunction([X[1], X[4], X[6], x[1]], [1., 1, 1, 1], 0.), MOI.EqualTo(1.))
+            c2 = MOI.addconstraint!(m, MOI.ScalarAffineFunction([X; x[2]; x[3]], [1., 2, 2, 1, 2, 1, 1, 1], 0.), MOI.EqualTo(1/2))
+
+
+            MOI.setobjective!(m, MOI.ScalarAffineFunction([X[1:2]; X[4:6]; x[1]], [2., 2, 0, 2, 2, 2, 1], 0.))
+
+
+            @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.ScalarAffineFunction{Float64}, EqualTo{Float64}}()) == 2
+            @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.VectorVariablewiseFunction, MOI.PositiveSemidefiniteConeTriangle}()) == 1
+            @test MOI.getattribute(m, MOI.NumberOfConstraints{MOI.VectorVariablewiseFunction, MOI.SecondOrderCone}()) == 1
+
+            MOI.optimize!(m)
+
+            @test MOI.cangetattribute(m, MOI.TerminationStatus())
+            @test MOI.getattribute(m, MOI.TerminationStatus()) == MOI.Success
+
+            @test MOI.cangetattribute(m, MOI.PrimalStatus())
+            @test MOI.getattribute(m, MOI.PrimalStatus()) == MOI.FeasiblePoint
+            @test MOI.cangetattribute(m, MOI.DualStatus())
+            @test MOI.getattribute(m, MOI.DualStatus()) == MOI.FeasiblePoint
+
+            @test MOI.cangetattribute(m, MOI.ObjectiveValue())
+            @test MOI.getattribute(m, MOI.ObjectiveValue()) ≈ 0.705710509 atol=ε
+
+            @test MOI.cangetattribute(m, MOI.VariablePrimal(), X)
+            Xv = MOI.getattribute(m, MOI.VariablePrimal(), X)
+            @test MOI.cangetattribute(m, MOI.VariablePrimal(), x)
+            xv = MOI.getattribute(m, MOI.VariablePrimal(), x)
+
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), c1)
+            y1 = MOI.getattribute(m, MOI.ConstraintDual(), c1)
+            @test MOI.cangetattribute(m, MOI.ConstraintDual(), c2)
+            y2 = MOI.getattribute(m, MOI.ConstraintDual(), c2)
+            y = [y1, y2]
+
+            #    X11 X21 X31 X22 X32 X33  x1  x2  x3
+            c = [  2,  2,  0,  2,  2,  2,  1,  0,  0]
+            A = [  1   0   0   1   0   1   1   0   0;
+                   1   2   2   1   2   1   0   1   1]
+            b = [1, 1/2]
+            # Check primal objective
+            comp_pobj = dot(c, [Xv; xv])
+            # Check dual objective
+            comp_dobj = -dot(y, b)
+            @test comp_pobj ≈ comp_dobj atol=ε
+
+            var = c + A' * y
```

Comment:
This might be a bit confusing for readers since it follows the old variable duals concepts which don't exist in MOI. Could we write this to be consistent with http://www.juliaopt.org/MathOptInterface.jl/latest/apimanual.html#Duals-1?